### PR TITLE
initial implementation for tracking musics and access roles

### PR DESCRIPTION
**Pull request by ikwuoz**
This implementation aims to keep track of music ownership on the platform, and some limited access control mechanisms for who can add music or add an artist

### CODE REVIEW

File: packages/hardhat/contracts/Soundly.sol [hunk 1]
```diff
@@ -0,0 +1,75 @@
+pragma solidity >=0.8.0 <0.9.0;
```
Comment (jonddon) [inline packages/hardhat/contracts/Soundly.sol]: let's stick to a version 0.8.4

File: packages/hardhat/contracts/Soundly.sol [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        bool isArtist;
+        bytes32[] musicsID;
+    }
+
+    mapping(address => Artist) artists;
+
+    // these two address parameters would be have admin priviledges
+    constructor(address _admin1, address _admin2) public {
+        _setRoleAdmin(ADMIN_ROLE, _msgSender());
+        _setRoleAdmin(ARTIST_ROLE, _msgSender());
```
Comment (jonddon) [inline packages/hardhat/contracts/Soundly.sol:29]: why are you granting msgSender two roles?

File: packages/hardhat/contracts/SoundlySuperApp.sol [hunk 3]
```diff
@@ -0,0 +1,6 @@
+pragma solidity >=0.8.0 <0.9.0;
```
Comment (jonddon) [inline packages/hardhat/contracts/SoundlySuperApp.sol]: use 0.8.4

**Review by jonddon — CHANGES_REQUESTED** [review]

File: packages/hardhat/contracts/Soundly.sol (hunk 1 above)
Comment (ikwuoz) [reply packages/hardhat/contracts/Soundly.sol]: Okay, would make adjustments

File: packages/hardhat/contracts/Soundly.sol (hunk 2 above)
Comment (ikwuoz) [reply packages/hardhat/contracts/Soundly.sol:29]: The oz access control use two levels:
There is an 'Admin' for every role, and then members with the 'role' itself.
So msgSender is the 'Admin' for the every role defined in the contract i.e the 'admin' role and the 'artist' role

File: packages/hardhat/contracts/SoundlySuperApp.sol (hunk 3 above)
Comment (ikwuoz) [reply packages/hardhat/contracts/SoundlySuperApp.sol]: I have made the changes

**Pull request merged**